Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -1427,28 +1427,90 @@ bool CStaticFunctionDefinitions::SetElementHealth(CClientEntity& Entity, float f
         {
             // Grab the model
             CClientPed& Ped = static_cast<CClientPed&>(Entity);
+            float       fCurHealth = Ped.GetHealth();
 
             // Limit to max health
             float fMaxHealth = Ped.GetMaxHealth();
             if (fHealth > fMaxHealth)
                 fHealth = fMaxHealth;
 
-            // Set the new health
+            if (fCurHealth == fHealth)
```

Comment:
This condition will almost always be `false`. Epsilon comparison should be used.